deploy/buildtools/buildafi.py

```diff
@@ -171,7 +189,7 @@ def on_build_failure():
 
     # if we're unlucky, multiple vivado builds may launch at the same time. so we
     # append the build node IP + a random string to diff them in s3
-    global_append = "-" + env.host_string + "-" + ''.join(random.SystemRandom().choice(string.ascii_uppercase + string.digits) for _ in range(10)) + ".tar"
+    global_append = "-" + str(env.host_string) + "-" + ''.join(random.SystemRandom().choice(string.ascii_uppercase + string.digits) for _ in range(10)) + ".tar"
```

Comment:
When using tar2afi, Fabric's env.host_string will be None.  explicitly casting it to str makes this avoid an exception and it should be fine for the host element to be empty for tar2afi.  To be more clear, we could check for None and use 'localhost' if that's preferable.